torch/nn/modules/module.py

```diff
@@ -2422,3 +2438,7 @@ def _replicate_for_data_parallel(self):
         replica._is_replica = True  # type: ignore[assignment]
 
         return replica
+
+    def compile(self, *args, **kwargs):
+        optimized_module = torch.compile(self, *args, **kwargs)
```

Comment:
Do you need to preserve the OptimizedModule here? Why not just compile the _call_impl and replace it?

Reply:
Ho interesting